fury/ui/elements.py

```diff
@@ -3484,3 +3486,236 @@ def left_button_dragged(self,  i_ren, _obj, element):
+    """
+
+    def __init__(self, position=(0, 0)):
+        """ Initialize the color picker.
+        Parameters
+        ----------
+        side : int
+            The side length of the square(and height of the hue bar) in pixels.
+        position : (float, float)
+                Coordinates (x, y) of the lower-left corner of the sqaure.
+        """
+        super(ColorPicker, self).__init__(position)
+        self.side = 100
+        self.pointer.position = self.ColorSelectionSquare.center
+        self.position = position
+        self.current_hue = 0
+        self.current_saturation = 0.5
+        self.current_value = 0.5
+        # Offer some standard hooks to the user.
+        self.on_change = lambda color_picker, r, g, b: None
+
+    def _setup(self):
+        """ Setup this UI component.
+        """
+        # Setup the hue bar
+        # this line was changed
+        self.HueBar = ScalarBarActor()
+        self.HueBar.GetPositionCoordinate().SetCoordinateSystemToDisplay()
+        self.HueBar.GetPosition2Coordinate().SetCoordinateSystemToDisplay()
+        self.HueBar.SetNumberOfLabels(0)
+        hueLut = LookupTable()  # this line was changed
+        hueLut.SetTableRange(0, 1)
+        hueLut.SetHueRange(0, 1)
+        hueLut.SetSaturationRange(1, 1)
+        hueLut.SetValueRange(1, 1)
+        hueLut.Build()
+        self.HueBar.SetLookupTable(hueLut)
+        self.add_callback(self.HueBar, "MouseMoveEvent",
+                          self.hue_select_callback)
+        # Setup colors for the square
+        colors = UnsignedCharArray()  # this line was changed
+        colors.SetNumberOfComponents(3)
+        colors.SetName("Colors")
+        colors.InsertNextTuple3(*self.hsv2rgb(0, 0, 0))
+        colors.InsertNextTuple3(*self.hsv2rgb(0, 1, 0))
+        colors.InsertNextTuple3(*self.hsv2rgb(0, 1, 1))
+        colors.InsertNextTuple3(*self.hsv2rgb(0, 0, 1))
+        # Setup the square
+        self.ColorSelectionSquare = Rectangle2D()
+        self.ColorSelectionSquare._polygonPolyData.GetPointData().\
+            SetScalars(colors)
+        # if int(VTK_VERSION) <= 5:
+        # self.ColorSelectionSquare._polygonPolyData.Update()
+        self.ColorSelectionSquare.on_left_mouse_button_dragged = \
+            self.color_select_callback
+        # Setup the circle pointer
+        self.pointer = Disk2D(outer_radius=1)
+
+    def _get_actors(self):
+        """ Get the actors composing this UI component.
+        """
+        return self.ColorSelectionSquare.actors + self.HueBar +\
+            self.pointer.actors
+
+    def _add_to_scene(self, scene):
+        """Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        scene : scene
+        """
+        self.ColorSelectionSquare._add_to_scene(scene)  # this line was changed
+        self.pointer._add_to_scene(scene)  # this line was changed
```

Comment:
Hello @saran820 ,
I am not sure if you are still working on it or not but I figured out why the interaction wasn't working for `Rectangle2D`. 

If you change these lines of code as suggested, the below code would work.
```suggestion
        self.ColorSelectionSquare.add_to_scene(scene)  # this line was changed
        self.pointer.add_to_scene(scene)  # this line was changed
```

Modified Code:
```python

from fury import ui, window
from fury.data import fetch_viz_icons
import sys
import numpy as np

cp = ui.ColorPicker(side=300, position=(0, 0))

rect = ui.Rectangle2D(size=(100, 100), position=(400, 400))


def change_rect_color(color_picker, r, g, b):
    rect.color = np.asarray(color_picker.hsv2rgb(color_picker.current_hue, color_picker.current_saturation,
                                                 color_picker.current_value)) / 255


cp.on_change = change_rect_color

current_size = (800, 800)
show_manager = window.ShowManager(size=current_size,
                                  title="Colorpicker on Rectangle")

show_manager.scene.add(cp)
show_manager.scene.add(rect)

interactive = True

if interactive:
    show_manager.start()

window.record(show_manager.scene, size=current_size,
              out_path="viz_colorpicker.png")


```